rectools/models/popular.py

```diff
@@ -41,17 +43,83 @@ class Popularity(Enum):
     SUM_WEIGHT = "sum_weight"
 
 
+def _serialize_timedelta(td: tp.Optional[tp.Union[None, dict, timedelta]]) -> tp.Optional[timedelta]:
+    if isinstance(td, dict):
+        return timedelta(
+            days=td.get("days", 0),
+            seconds=td.get("seconds", 0),
+            microseconds=td.get("microseconds", 0),
+            milliseconds=td.get("milliseconds", 0),
+            minutes=td.get("minutes", 0),
+            hours=td.get("hours", 0),
+            weeks=td.get("weeks", 0),
+        )
+    return td
+
+
+def _deserialize_timedelta(td: tp.Optional[timedelta]) -> tp.Optional[dict]:
+    if td is None:
+        return td
+    return {"days": td.days, "seconds": td.seconds, "microseconds": td.microseconds}
+
+
+TimeDelta = tpe.Annotated[
+    tp.Union[None, timedelta, dict],
```

Comment:
1. No, it can't be `dict`. If it can be a dict, then you don't need serializers and deserializers.  
2. You don't need to care here (and in your functions) about `None`. `pydantic` will do it for you later. Here you just `timedelta`, and later in config `tp.Optional[TimeDelta]`

Reply:
Fixed typing.